src/xdp-utils.c

```diff
@@ -144,11 +145,72 @@ xdp_app_info_new (XdpAppInfoKind kind)
   return app_info;
 }
 
+void
+decode_hex_escapes (char *str[])
+{
+  char hex[] = "0xhh";
+
+  for (int i=0; (*str)[i]!='\0' && (*str)[i+1]!='\0'; ++i)
+  {
+    if ((*str)[i] == '\\' && (*str)[i+1] == 'x')
+      {
+        hex[2] = (*str)[i+2];
+        hex[3] = (*str)[i+3];
+        char unescaped = (char) strtol (hex, NULL, 16);
+        (*str)[i+2] = '\\';
+        (*str)[i+3] = unescaped;
+        i += 3;
+      }
+  }
+  char **splits = g_strsplit(*str, "\\x\\", -1);
+  g_free (*str);
+  *str = g_strjoinv ("", splits);
+  g_strfreev (splits);
+  return;
+}
+
+void
+get_app_id_from_pid (pid_t pid, XdpAppInfo *app_info)
+{
+  char *unit, **dash_splits, **at_splits, **app_id;
+  int res, len;
+
+  res = sd_pid_get_user_unit(pid, &unit);
+  /*
+   * the session might not be managed by systemd
+   * or there could be error fetching own systemd units
+   * or the unit might not be started by the the desktop environment (e.g. it's a script run from terminal)
+   */
```

Comment:
I don't think this comment is necessary - probably better to mention these cases in the commit message?

Reply:
Seems like a helpful comment to me